### PR TITLE
includes: C++ compatibility and consistent protection #defines

### PR DESCRIPTION
**Pull request by chrissie-c**
Include a test in "make check" that makes sure all headers can
be included in their own, have unique and consistent #define
to protect them, and that they work with C++ code

[THIS RUN MAINLY FOR CI TESTING]

### CODE REVIEW

File: tests/test-headers.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	continue
+    fi
+    PROTECT="PCMK__`echo $NAME | tr '[:lower:]' '[:upper:]' | tr '/\.' '_' |tr '-' '_' `"
+    PROTECT=`echo $PROTECT|sed 's/_H/__H/'`  # PCMK uses double underscore for .h (eg NAME__H)
+
+    echo "#include <$NAME>" > _test_file.c
+    echo "#ifndef $PROTECT" >> _test_file.c
+    echo "#error no header protector in file $i" >> _test_file.c
+    echo "#endif" >> _test_file.c
+    echo "int main(void) {return 0;}" >> _test_file.c
```
Comment (kgaillot) [inline tests/test-headers.sh:25]: might as well throw _test_file.c into .gitignore in case this gets interrupted

File: tests/test-headers.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/bin.sh
```
Comment (kgaillot) [inline tests/test-headers.sh:1]: slash :)

**Review by kgaillot — COMMENTED** [review]
nice, before you know it we're going to look like we know what we're doing :)

a first look ...

**Comment by chrissie-c** [comment]
The covscan failures bother me. Did I fork at a bad time, or have I somehow subtly broken something ?

**Comment by chrissie-c** [comment]
Ahh good, a rebase to latest "main" seems to have got rid of the covscan errors. *phew*

**Comment by wenningerk** [comment]
maybe we could then toss the stuff with at least partly similar intention in .../maint/Makefile.am.

**Comment by kgaillot** [comment]
> maybe we could then toss the stuff with at least partly similar intention in .../maint/Makefile.am.

oh right, i forgot this idea started as a more generic replacement for that

**Comment by wenningerk** [comment]
> > maybe we could then toss the stuff with at least partly similar intention in .../maint/Makefile.am.
> 
> oh right, i forgot this idea started as a more generic replacement for that

One thing to scratch from the release checklist ;-)
Similar probably for the coverity-stuff. But in this case we should keep what is needed + howto as to be able to trigger a valgrind-run without CI.

**Comment by chrissie-c** [comment]
@kgaillot I'm happy with it if you are

**Pull request merged**